src/portable/link.rs

```diff
@@ -198,32 +213,43 @@ pub fn link(cmd: &Link, opts: &Options) -> anyhow::Result<()> {
         }
     );
     let mut config = config.with_cert_verifier(verifier.clone());
-    let connect_result = connect(&config);
-    if let Err(e) = connect_result {
-        if e.is::<PasswordRequired>() {
-            let password;
-
-            if opts.conn_options.password_from_stdin {
-                password = tty_password::read_stdin()?
-            } else if !cmd.non_interactive {
-                password = tty_password::read(format!(
+    let mut conn = connect(&config).await;
+
+    if let Err(e) = conn {
+        if let Some(err) = e.downcast_ref::<edgedb_tokio::Error>() {
```

Comment:
I would suggest not to bring Rust async into this fight :D It I very convenient in some use cases, but I think those use cases don't apply to simple CLI app that is running synchronously.

So my suggestion is to add `#[tokio::main]` to `connect()` & `ask_branch()` and to keep `link()` sync.

Reply:
In #1286, I've replaced all `#[tokio::main]` with `#[tokio::main(flavor = "current_thread")]`, which should be done here as well. It might even solve the issue with "shutdown context".